File: static/js/strategies.js
// Strategies page JavaScript functionality

let strategiesData = [];
let currentPage = 1;
const itemsPerPage = 10;

// Initialize strategies page
document.addEventListener('DOMContentLoaded', function() {
    loadStrategies();
    
    // Auto-refresh every 30 seconds
    setInterval(loadStrategies, 30000);
});

// Load all strategies
async function loadStrategies() {
    try {
        const response = await fetch('/api/v1/strategies/');
        strategiesData = await response.json();
        
        updateStrategyCounts();
        displayStrategies();
        
    } catch (error) {
        console.error('Error loading strategies:', error);
        showAlert('Error loading strategies', 'danger');
    }
}

// Update strategy count cards
function updateStrategyCounts() {
    const total = strategiesData.length;
    const active = strategiesData.filter(s => s.status === 'active').length;
    const paused = strategiesData.filter(s => s.status === 'paused').length;
    const stopped = strategiesData.filter(s => s.status === 'stopped').length;
    
    document.getElementById('total-strategies').textContent = total;
    document.getElementById('active-strategies').textContent = active;
    document.getElementById('paused-strategies').textContent = paused;
    document.getElementById('stopped-strategies').textContent = stopped;
}

// Display strategies in table
function displayStrategies(filteredData = null) {
    const data = filteredData || strategiesData;
    const tableBody = document.getElementById('strategies-table-body');
    
    if (data.length === 0) {
        tableBody.innerHTML = `
            <tr>
                <td colspan="8" class="text-center text-muted">
                    <i class="fas fa-cog fa-2x mb-3 opacity-50"></i>
                    <p>No strategies found</p>
                </td>
            </tr>
        `;
        return;
    }
    
    let html = '';
    
    data.forEach(strategy => {
        const statusClass = getStatusBadgeClass(strategy.status);
        const statusIcon = getStatusIcon(strategy.status);
        
        html += `
            <tr>
                <td>
                    <div class="d-flex align-items-center">
                        <strong>${strategy.name}</strong>
                        ${strategy.description ? `<small class="text-muted ms-2">${strategy.description.substring(0, 50)}...</small>` : ''}
                    </div>
                </td>
                <td>
                    <span class="badge bg-info">${strategy.strategy_type}</span>
                </td>
                <td>
                    <span class="fw-bold">${strategy.symbol}</span>
                </td>
                <td>
                    <span class="badge ${statusClass}">
                        <i class="${statusIcon} me-1"></i>${strategy.status}
                    </span>
                </td>
                <td>${strategy.initial_lot_size}</td>
                <td>${strategy.recovery_step} pips</td>
                <td>
                    <small>${new Date(strategy.created_at).toLocaleDateString()}</small>
                </td>
                <td>
                    <div class="btn-group btn-group-sm" role="group">
                        ${strategy.status === 'active' ? 
                            `<button class="btn btn-outline-warning" onclick="pauseStrategy(${strategy.id})" title="Pause">
                                <i class="fas fa-pause"></i>
                            </button>
                            <button class="btn btn-outline-danger" onclick="stopStrategy(${strategy.id})" title="Stop">
                                <i class="fas fa-stop"></i>
                            </button>` :
                            `<button class="btn btn-outline-success" onclick="startStrategy(${strategy.id})" title="Start">
                                <i class="fas fa-play"></i>
                            </button>`
                        }
                        <button class="btn btn-outline-primary" onclick="viewStrategyDetails(${strategy.id})" title="Details">
                            <i class="fas fa-eye"></i>
                        </button>
                        <button class="btn btn-outline-secondary" onclick="editStrategy(${strategy.id})" title="Edit" ${strategy.status === 'active' ? 'disabled' : ''}>
                            <i class="fas fa-edit"></i>
                        </button>
                        <button class="btn btn-outline-danger" onclick="deleteStrategy(${strategy.id})" title="Delete" ${strategy.status === 'active' ? 'disabled' : ''}>
                            <i class="fas fa-trash"></i>
                        </button>
                    </div>
                </td>
            </tr>
        `;
    });
    
    tableBody.innerHTML = html;
}

// Filter strategies
function filterStrategies() {
    const statusFilter = document.getElementById('statusFilter').value;
    const typeFilter = document.getElementById('typeFilter').value;
    
    let filteredData = strategiesData;
    
    if (statusFilter) {
        filteredData = filteredData.filter(s => s.status === statusFilter);
    }
    
    if (typeFilter) {
        filteredData = filteredData.filter(s => s.strategy_type === typeFilter);
    }
    
    displayStrategies(filteredData);
}

// Start strategy
async function startStrategy(strategyId) {
    const sessionName = prompt('Enter session name (optional):') || `Session_${Date.now()}`;
    
    try {
        showLoading();
        const response = await fetch(`/api/v1/strategies/${strategyId}/start`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ session_name: sessionName })
        });
        
        const result = await response.json();
        
        if (response.ok) {
            showAlert(result.message, 'success');
            await loadStrategies();
        } else {
            showAlert(result.detail || 'Failed to start strategy', 'danger');
        }
    } catch (error) {
        console.error('Error starting strategy:', error);
        showAlert('Error starting strategy', 'danger');
    } finally {
        hideLoading();
    }
}

// Stop strategy
async function stopStrategy(strategyId) {
    if (!confirm('Are you sure you want to stop this strategy?')) {
        return;
    }
    
    try {
        showLoading();
        const response = await fetch(`/api/v1/strategies/${strategyId}/stop`, {
            method: 'POST'
        });
        
        const result = await response.json();
        
        if (response.ok) {
            showAlert(result.message, 'success');
            await loadStrategies();
        } else {
            showAlert(result.detail || 'Failed to stop strategy', 'danger');
        }
    } catch (error) {
        console.error('Error stopping strategy:', error);
        showAlert('Error stopping strategy', 'danger');
    } finally {
        hideLoading();
    }
}

// Pause strategy (placeholder - would need backend implementation)
async function pauseStrategy(strategyId) {
    showAlert('Pause functionality not yet implemented', 'info');
}

// Create strategy
async function createStrategy() {
    const form = document.getElementById('createStrategyForm');
    if (!form.checkValidity()) {
        form.reportValidity();
        return;
    }
    
    const strategyData = {
        name: document.getElementById('strategyName').value,
        strategy_type: document.getElementById('strategyType').value,
        symbol: document.getElementById('strategySymbol').value,
        description: document.getElementById('description').value,
        initial_lot_size: parseFloat(document.getElementById('initialLotSize').value),
        max_lot_size: parseFloat(document.getElementById('maxLotSize').value),
        recovery_step: parseFloat(document.getElementById('recoveryStep').value),
        max_trades: parseInt(document.getElementById('maxTrades').value),
        take_profit: parseFloat(document.getElementById('takeProfit').value),
        stop_loss: parseFloat(document.getElementById('stopLoss').value),
        max_drawdown: parseFloat(document.getElementById('maxDrawdown').value),
        risk_per_trade: parseFloat(document.getElementById('riskPerTrade').value)
    };
    
    try {
        showLoading();
        const response = await fetch('/api/v1/strategies/', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(strategyData)
        });
        
        const result = await response.json();
        
        if (response.ok) {
            showAlert('Strategy created successfully!', 'success');
            bootstrap.Modal.getInstance(document.getElementById('createStrategyModal')).hide();
            form.reset();
            await loadStrategies();
        } else {
            showAlert(result.detail || 'Failed to create strategy', 'danger');
        }
    } catch (error) {
        console.error('Error creating strategy:', error);
        showAlert('Error creating strategy', 'danger');
    } finally {
        hideLoading();
    }
}

// Edit strategy
async function editStrategy(strategyId) {
    try {
        const response = await fetch(`/api/v1/strategies/${strategyId}`);
        const strategy = await response.json();
        
        if (response.ok) {
            // Populate edit form
            document.getElementById('editStrategyId').value = strategy.id;
            document.getElementById('editStrategyName').value = strategy.name;
            // Add more fields as needed
            
            const modal = new bootstrap.Modal(document.getElementById('editStrategyModal'));
            modal.show();
        } else {
            showAlert('Failed to load strategy details', 'danger');
        }
    } catch (error) {
        console.error('Error loading strategy for edit:', error);
        showAlert('Error loading strategy', 'danger');
    }
}

// Update strategy
async function updateStrategy() {
    const strategyId = document.getElementById('editStrategyId').value;
    const form = document.getElementById('editStrategyForm');
    
    if (!form.checkValidity()) {
        form.reportValidity();
        return;
    }
    
    const updateData = {
        name: document.getElementById('editStrategyName').value
        // Add more fields as needed
    };
    
    try {
        showLoading();
        const response = await fetch(`/api/v1/strategies/${strategyId}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(updateData)
        });
        
        const result = await response.json();
        
        if (response.ok) {
            showAlert('Strategy updated successfully!', 'success');
            bootstrap.Modal.getInstance(document.getElementById('editStrategyModal')).hide();
            await loadStrategies();
        } else {
            showAlert(result.detail || 'Failed to update strategy', 'danger');
        }
    } catch (error) {
        console.error('Error updating strategy:', error);
        showAlert('Error updating strategy', 'danger');
    } finally {
        hideLoading();
    }
}

// Delete strategy
async function deleteStrategy(strategyId) {
    const strategy = strategiesData.find(s => s.id === strategyId);
    if (!strategy) return;
    
    if (!confirm(`Are you sure you want to delete strategy "${strategy.name}"? This action cannot be undone.`)) {
        return;
    }
    
    try {
        showLoading();
        const response = await fetch(`/api/v1/strategies/${strategyId}`, {
            method: 'DELETE'
        });
        
        const result = await response.json();
        
        if (response.ok) {
            showAlert(result.message, 'success');
            await loadStrategies();
        } else {
            showAlert(result.detail || 'Failed to delete strategy', 'danger');
        }
    } catch (error) {
        console.error('Error deleting strategy:', error);
        showAlert('Error deleting strategy', 'danger');
    } finally {
        hideLoading();
    }
}

// View strategy details
async function viewStrategyDetails(strategyId) {
    try {
        showLoading();
        
        // Load strategy details
        const strategyResponse = await fetch(`/api/v1/strategies/${strategyId}`);
        const strategy = await strategyResponse.json();
        
        // Load strategy sessions
        const sessionsResponse = await fetch(`/api/v1/strategies/${strategyId}/sessions`);
        const sessions = await sessionsResponse.json();
        
        if (strategyResponse.ok && sessionsResponse.ok) {
            displayStrategyDetails(strategy, sessions);
            
            const modal = new bootstrap.Modal(document.getElementById('strategyDetailsModal'));
            modal.show();
        } else {
            showAlert('Failed to load strategy details', 'danger');
        }
    } catch (error) {
        console.error('Error loading strategy details:', error);
        showAlert('Error loading strategy details', 'danger');
    } finally {
        hideLoading();
    }
}

// Display strategy details in modal
function displayStrategyDetails(strategy, sessions) {
    document.getElementById('strategyDetailsTitle').textContent = `Strategy: ${strategy.name}`;
    
    const content = `
        <div class="row">
            <div class="col-md-6">
                <h6 class="border-bottom pb-2">Strategy Information</h6>
                <table class="table table-sm">
                    <tr><td><strong>Name:</strong></td><td>${strategy.name}</td></tr>
                    <tr><td><strong>Type:</strong></td><td><span class="badge bg-info">${strategy.strategy_type}</span></td></tr>
                    <tr><td><strong>Symbol:</strong></td><td>${strategy.symbol}</td></tr>
                    <tr><td><strong>Status:</strong></td><td><span class="badge ${getStatusBadgeClass(strategy.status)}">${strategy.status}</span></td></tr>
                    <tr><td><strong>Created:</strong></td><td>${new Date(strategy.created_at).toLocaleString()}</td></tr>
                </table>
                
                <h6 class="border-bottom pb-2 mt-4">Trading Parameters</h6>
                <table class="table table-sm">
                    <tr><td><strong>Initial Lot Size:</strong></td><td>${strategy.initial_lot_size}</td></tr>
                    <tr><td><strong>Max Lot Size:</strong></td><td>${strategy.max_lot_size}</td></tr>
                    <tr><td><strong>Recovery Step:</strong></td><td>${strategy.recovery_step} pips</td></tr>
                    <tr><td><strong>Max Trades:</strong></td><td>${strategy.max_trades}</td></tr>
                    <tr><td><strong>Take Profit:</strong></td><td>${strategy.take_profit} pips</td></tr>
                    <tr><td><strong>Stop Loss:</strong></td><td>${strategy.stop_loss} pips</td></tr>
                    <tr><td><strong>Max Drawdown:</strong></td><td>$${strategy.max_drawdown}</td></tr>
                </table>
            </div>
            
            <div class="col-md-6">
                <h6 class="border-bottom pb-2">Trading Sessions</h6>
                <div style="max-height: 400px; overflow-y: auto;">
                    ${sessions.length === 0 ? 
                        '<p class="text-muted">No trading sessions found</p>' :
                        sessions.map(session => `
                            <div class="border rounded p-3 mb-3">
                                <div class="d-flex justify-content-between align-items-start mb-2">
                                    <h6 class="mb-0">${session.session_name}</h6>
                                    <span class="badge ${session.status === 'active' ? 'bg-success' : 'bg-secondary'}">${session.status}</span>
                                </div>
                                <small class="text-muted">
                                    Started: ${new Date(session.start_time).toLocaleString()}<br>
                                    ${session.end_time ? `Ended: ${new Date(session.end_time).toLocaleString()}` : 'Still running'}
                                </small>
                                <div class="row mt-2">
                                    <div class="col-6"><small>Total Trades: <strong>${session.total_trades}</strong></small></div>
                                    <div class="col-6"><small>Win Rate: <strong>${session.total_trades > 0 ? ((session.winning_trades / session.total_trades) * 100).toFixed(1) : 0}%</strong></small></div>
                                    <div class="col-12"><small>P&L: <strong class="${session.total_profit_loss >= 0 ? 'text-success' : 'text-danger'}">$${session.total_profit_loss.toFixed(2)}</strong></small></div>
                                </div>
                            </div>
                        `).join('')
                    }
                </div>
            </div>
        </div>
    `;
    
    document.getElementById('strategyDetailsContent').innerHTML = content;
}

// Show create strategy modal
function showCreateStrategyModal() {
    const modal = new bootstrap.Modal(document.getElementById('createStrategyModal'));
    modal.show();
}

// Helper functions
function getStatusBadgeClass(status) {
    const classes = {
        'active': 'bg-success',
        'inactive': 'bg-secondary',
        'paused': 'bg-warning',
        'stopped': 'bg-danger'
    };
    return classes[status] || 'bg-secondary';
}

function getStatusIcon(status) {
    const icons = {
        'active': 'fas fa-play',
        'inactive': 'fas fa-pause',
        'paused': 'fas fa-pause-circle',
        'stopped': 'fas fa-stop'
    };
    return icons[status] || 'fas fa-question';
}

// Export functions
window.Strategies = {
    loadStrategies,
    startStrategy,
    stopStrategy,
    createStrategy,
    deleteStrategy,
    viewStrategyDetails
};

File: app/routers/strategies.py
"""
Strategy management endpoints for creating, managing, and controlling trading strategies
"""

from fastapi import APIRouter, Depends, HTTPException, BackgroundTasks
from sqlalchemy.orm import Session
from typing import List, Optional, Dict, Any
from pydantic import BaseModel
from datetime import datetime

from app.database.database import get_db
from app.models.models import Strategy, TradingSession, StrategyStatus
from app.services.martingale_service import MartingaleService
from app.core.config import settings

router = APIRouter()

# Pydantic models
class StrategyCreateModel(BaseModel):
    name: str
    description: Optional[str] = None
    strategy_type: str = "martingale"
    symbol: str
    initial_lot_size: float
    max_lot_size: float = 1.0
    recovery_step: float = 50.0
    take_profit: float = 100.0
    stop_loss: float = 500.0
    max_trades: int = 5
    max_drawdown: float = 1000.0
    risk_per_trade: float = 2.0

class StrategyResponseModel(BaseModel):
    id: int
    name: str
    description: Optional[str]
    strategy_type: str
    status: str
    symbol: str
    initial_lot_size: float
    max_lot_size: float
    recovery_step: float
    take_profit: float
    stop_loss: float
    max_trades: int
    max_drawdown: float
    risk_per_trade: float
    created_at: datetime
    updated_at: Optional[datetime]
    
    class Config:
        from_attributes = True

class StrategyUpdateModel(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    symbol: Optional[str] = None
    initial_lot_size: Optional[float] = None
    max_lot_size: Optional[float] = None
    recovery_step: Optional[float] = None
    take_profit: Optional[float] = None
    stop_loss: Optional[float] = None
    max_trades: Optional[int] = None
    max_drawdown: Optional[float] = None
    risk_per_trade: Optional[float] = None

class TradingSessionModel(BaseModel):
    id: int
    strategy_id: int
    session_name: str
    start_time: datetime
    end_time: Optional[datetime]
    status: str
    total_trades: int
    winning_trades: int
    losing_trades: int
    total_profit_loss: float
    max_drawdown_reached: float
    
    class Config:
        from_attributes = True

@router.post("/", response_model=StrategyResponseModel, summary="Create a new strategy")
async def create_strategy(strategy: StrategyCreateModel, db: Session = Depends(get_db)):
    """Create a new trading strategy"""
    
    # Check if strategy name already exists
    existing_strategy = db.query(Strategy).filter(Strategy.name == strategy.name).first()
    if existing_strategy:
        raise HTTPException(status_code=400, detail="Strategy name already exists")
    
    # Create new strategy
    db_strategy = Strategy(
        name=strategy.name,
        description=strategy.description,
        strategy_type=strategy.strategy_type,
        symbol=strategy.symbol.upper(),
        initial_lot_size=strategy.initial_lot_size,
        max_lot_size=strategy.max_lot_size,
        recovery_step=strategy.recovery_step,
        take_profit=strategy.take_profit,
        stop_loss=strategy.stop_loss,
        max_trades=strategy.max_trades,
        max_drawdown=strategy.max_drawdown,
        risk_per_trade=strategy.risk_per_trade,
        status=StrategyStatus.INACTIVE
    )
    
    db.add(db_strategy)
    db.commit()
    db.refresh(db_strategy)
    
    return StrategyResponseModel(
        id=db_strategy.id,
        name=db_strategy.name,
        description=db_strategy.description,
        strategy_type=db_strategy.strategy_type,
        status=db_strategy.status.value,
        symbol=db_strategy.symbol,
        initial_lot_size=db_strategy.initial_lot_size,
        max_lot_size=db_strategy.max_lot_size,
        recovery_step=db_strategy.recovery_step,
        take_profit=db_strategy.take_profit,
        stop_loss=db_strategy.stop_loss,
        max_trades=db_strategy.max_trades,
        max_drawdown=db_strategy.max_drawdown,
        risk_per_trade=db_strategy.risk_per_trade,
        created_at=db_strategy.created_at,
        updated_at=db_strategy.updated_at
    )

@router.get("/", response_model=List[StrategyResponseModel], summary="Get all strategies")
async def get_strategies(
    status: Optional[str] = None,
    strategy_type: Optional[str] = None,
    db: Session = Depends(get_db)
):
    """Get all strategies with optional filtering"""
    
    query = db.query(Strategy)
    
    if status:
        try:
            status_enum = StrategyStatus(status.lower())
            query = query.filter(Strategy.status == status_enum)
        except ValueError:
            raise HTTPException(status_code=400, detail=f"Invalid status: {status}")
    
    if strategy_type:
        query = query.filter(Strategy.strategy_type == strategy_type.lower())
    
    strategies = query.all()
    
    return [
        StrategyResponseModel(
            id=s.id,
            name=s.name,
            description=s.description,
            strategy_type=s.strategy_type,
            status=s.status.value,
            symbol=s.symbol,
            initial_lot_size=s.initial_lot_size,
            max_lot_size=s.max_lot_size,
            recovery_step=s.recovery_step,
            take_profit=s.take_profit,
            stop_loss=s.stop_loss,
            max_trades=s.max_trades,
            max_drawdown=s.max_drawdown,
            risk_per_trade=s.risk_per_trade,
            created_at=s.created_at,
            updated_at=s.updated_at
        )
        for s in strategies
    ]

@router.get("/{strategy_id}", response_model=StrategyResponseModel, summary="Get strategy by ID")
async def get_strategy(strategy_id: int, db: Session = Depends(get_db)):
    """Get a specific strategy by ID"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    return StrategyResponseModel(
        id=strategy.id,
        name=strategy.name,
        description=strategy.description,
        strategy_type=strategy.strategy_type,
        status=strategy.status.value,
        symbol=strategy.symbol,
        initial_lot_size=strategy.initial_lot_size,
        max_lot_size=strategy.max_lot_size,
        recovery_step=strategy.recovery_step,
        take_profit=strategy.take_profit,
        stop_loss=strategy.stop_loss,
        max_trades=strategy.max_trades,
        max_drawdown=strategy.max_drawdown,
        risk_per_trade=strategy.risk_per_trade,
        created_at=strategy.created_at,
        updated_at=strategy.updated_at
    )

@router.put("/{strategy_id}", response_model=StrategyResponseModel, summary="Update strategy")
async def update_strategy(
    strategy_id: int, 
    strategy_update: StrategyUpdateModel, 
    db: Session = Depends(get_db)
):
    """Update an existing strategy"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    # Check if strategy is active
    if strategy.status == StrategyStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Cannot update active strategy. Stop it first.")
    
    # Update fields
    update_data = strategy_update.dict(exclude_unset=True)
    for field, value in update_data.items():
        if hasattr(strategy, field):
            if field == 'symbol' and value:
                setattr(strategy, field, value.upper())
            else:
                setattr(strategy, field, value)
    
    strategy.updated_at = datetime.now()
    
    db.commit()
    db.refresh(strategy)
    
    return StrategyResponseModel(
        id=strategy.id,
        name=strategy.name,
        description=strategy.description,
        strategy_type=strategy.strategy_type,
        status=strategy.status.value,
        symbol=strategy.symbol,
        initial_lot_size=strategy.initial_lot_size,
        max_lot_size=strategy.max_lot_size,
        recovery_step=strategy.recovery_step,
        take_profit=strategy.take_profit,
        stop_loss=strategy.stop_loss,
        max_trades=strategy.max_trades,
        max_drawdown=strategy.max_drawdown,
        risk_per_trade=strategy.risk_per_trade,
        created_at=strategy.created_at,
        updated_at=strategy.updated_at
    )

@router.delete("/{strategy_id}", summary="Delete strategy")
async def delete_strategy(strategy_id: int, db: Session = Depends(get_db)):
    """Delete a strategy"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    if strategy.status == StrategyStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Cannot delete active strategy. Stop it first.")
    
    db.delete(strategy)
    db.commit()
    
    return {"message": f"Strategy '{strategy.name}' deleted successfully"}

@router.post("/{strategy_id}/start", summary="Start strategy")
async def start_strategy(
    strategy_id: int, 
    session_name: Optional[str] = None,
    background_tasks: BackgroundTasks = BackgroundTasks(),
    db: Session = Depends(get_db)
):
    """Start a trading strategy"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    if strategy.status == StrategyStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Strategy is already active")
    
    # Create new trading session
    session = TradingSession(
        strategy_id=strategy_id,
        session_name=session_name or f"{strategy.name}_{datetime.now().strftime('%Y%m%d_%H%M%S')}",
        status="active"
    )
    
    db.add(session)
    db.commit()
    db.refresh(session)
    
    # Update strategy status
    strategy.status = StrategyStatus.ACTIVE
    db.commit()
    
    # Start strategy in background
    if strategy.strategy_type == "martingale":
        martingale_service = MartingaleService()
        background_tasks.add_task(
            martingale_service.run_strategy,
            strategy_id=strategy_id,
            session_id=session.id
        )
    
    return {
        "message": f"Strategy '{strategy.name}' started successfully",
        "session_id": session.id,
        "session_name": session.session_name
    }

@router.post("/{strategy_id}/stop", summary="Stop strategy")
async def stop_strategy(strategy_id: int, db: Session = Depends(get_db)):
    """Stop a running strategy"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    if strategy.status != StrategyStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Strategy is not active")
    
    # Update strategy status
    strategy.status = StrategyStatus.STOPPED
    
    # Close active session
    active_session = db.query(TradingSession).filter(
        TradingSession.strategy_id == strategy_id,
        TradingSession.status == "active"
    ).first()
    
    if active_session:
        active_session.status = "stopped"
        active_session.end_time = datetime.now()
    
    db.commit()
    
    return {"message": f"Strategy '{strategy.name}' stopped successfully"}

@router.get("/{strategy_id}/sessions", response_model=List[TradingSessionModel], summary="Get strategy sessions")
async def get_strategy_sessions(strategy_id: int, db: Session = Depends(get_db)):
    """Get all trading sessions for a strategy"""
    
    strategy = db.query(Strategy).filter(Strategy.id == strategy_id).first()
    if not strategy:
        raise HTTPException(status_code=404, detail="Strategy not found")
    
    sessions = db.query(TradingSession).filter(
        TradingSession.strategy_id == strategy_id
    ).order_by(TradingSession.start_time.desc()).all()
    
    return [
        TradingSessionModel(
            id=session.id,
            strategy_id=session.strategy_id,
            session_name=session.session_name,
            start_time=session.start_time,
            end_time=session.end_time,
            status=session.status,
            total_trades=session.total_trades,
            winning_trades=session.winning_trades,
            losing_trades=session.losing_trades,
            total_profit_loss=session.total_profit_loss,
            max_drawdown_reached=session.max_drawdown_reached
        )
        for session in sessions
    ]